examples/md-flexible/CMakeLists.txt

```diff
@@ -95,27 +95,28 @@ add_test(
         CONFIGURATIONS checkExamples
 )
 
+# The AVX example should only be tested if AVX is available.
 set(HAS_AVX true)
 if (NOT AUTOPAS_USE_VECTORIZATION)
+    # If no vectorization is used, don't enable the example.
     message(STATUS "No vectorization used, not adding md-flexible-avx.test-unaligned to ctest.")
     set(HAS_AVX false)
-elseif(AUTOPAS_VECTOR_INSTRUCTIONS MATCHES "SSE")
+elseif (AUTOPAS_VECTOR_INSTRUCTIONS MATCHES "SSE")
+    # If vectorization is set to SSE: don't enable the example
     message(STATUS "Only SSE specified, not adding md-flexible-avx.test-unaligned to ctest.")
     set(HAS_AVX false)
-elseif(AUTOPAS_VECTOR_INSTRUCTIONS MATCHES "NATIVE")
+elseif (AUTOPAS_VECTOR_INSTRUCTIONS MATCHES "NATIVE")
+    # If Vectorization is set to native: we try to figure out the vectorization level.
     message(STATUS "Native vectorization level, trying to detect AVX capability.")
-    if(UNIX)
-        message(STATUS "Unix")
-        EXEC_PROGRAM(cat ARGS "/proc/cpuinfo" OUTPUT_VARIABLE CPUINFO)
-        STRING(REGEX REPLACE "^.*(avx).*$" "\\1" AVX_THERE ${CPUINFO})
-        STRING(COMPARE EQUAL "avx" "${AVX_THERE}" HAS_AVX)
-        message(STATUS "detected avx on Unix? ${HAS_AVX}")
-    else()
-        message(STATUS "no Unix")
-        # set to false, as we don't know...
-        set(HAS_AVX false)
-    endif()
-else()
+    try_compile(
+            HAS_AVX
+            ${CMAKE_BINARY_DIR}
+            ${AUTOPAS_SOURCE_DIR}/cmake/tests/has_avx_test.cpp
+            # We are abusing COMPILE_DEFINITIONS here. There is no other sane way of passing "-march=native" ...
+            COMPILE_DEFINITIONS "-march=native"
```

Comment:
found similar thing on `apple.com` so passing this using COMPILE_DEFINITIONS seems right...

https://opensource.apple.com/source/clang/clang-700.1.81/src/projects/compiler-rt/cmake/config-ix.cmake.auto.html